### PR TITLE
Updates to OSToy + ACK capability for S3 service integration

### PR DESCRIPTION
**Pull request by 0kashi**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -24,6 +25,9 @@ app.set('views', __dirname + '/views');
 app.set('view engine', 'ejs');
 app.set('port', process.env.PORT || 8080);
 
+// Set the region
+AWS.config.update({region: 'us-west-2'});  //TODO change to envvar
```
Comment (wgordon17) [inline app.js]: Did you want to change this before merging?

File: README.md [hunk 2]
```diff
@@ -33,6 +34,8 @@ the ConfigMap object
 the Secret object
 - **PERSISTENT_DIRECTORY** (*default: /var/demo\_files*): The fully-qualified path to the directory mounted
 with the PersistentVolume
+- **CLOUD**: The cloud that this app is running on (ex: AWS).
```
Comment (wgordon17) [inline README.md]: Just wondering if you wanted to try auto-detecting this with something like https://stackoverflow.com/a/60385093/6758654 ¯\\\_(ツ)_/¯

File: app.js [hunk 3]
```diff
@@ -259,6 +270,88 @@ app.get('/env-variables', function(request, response) {
...
+  AWS CONTROLLER FOR KUBERNETES
+*/
+
+//returns the object keys (names) that are in the bucket
+app.get('/ack', function(request, response) {
+  if (cloud !== "AWS") {
+    response.render('error', {'msg': 'Wrong cloud platform. In order to use the ACK this must be run on AWS.'});
+  } else {
+    // Create S3 service object
+    s3 = new AWS.S3({apiVersion: '2006-03-01'});
```
Comment (wgordon17) [inline app.js:302]: We're you able to get this running without credentials?

File: app.js [hunk 4]
```diff
@@ -259,6 +270,88 @@ app.get('/env-variables', function(request, response) {
...
+        // response.type(data.ContentType); // Set FileType
+        // response.send(data.Body);        // Send File Buffer
+        response.render('s3viewfile', {'filename': filename, 'file': data.Body});
+      }
+    });
+  }
+});
+
+//Uploading a file to s3
+app.post('/s3upload', function(request, response) {
```
Comment (wgordon17) [inline app.js:349]: This looks pretty cool! 👍

File: views/filesystem.ejs [hunk 5]
```diff
@@ -20,12 +20,12 @@
...
 
     <div class="row">
-        <div class="col-lg-6">
+        <div class="col-lg-9">
             <p>
                 Pods and Containers in OpenShift are ephemeral, meaning that when they terminate, any changes that
                 may have occurred, including changes to the filesystem, are lost. This is intended behavior, and
                 the way to ensure your filesystem changes are preserved is to rely on
-                <a href="https://docs.openshift.com/dedicated/4/storage/understanding-persistent-storage.html" target="_blank">
+                <a href="https://docs.openshift.com/rosa/storage/persistent_storage/rosa-persistent-storage-aws-ebs.html" target="_blank">
```
Comment (wgordon17) [inline views/filesystem.ejs]: Is this the right docs page for, say, someone testing on OSD on GCP? ¯\\\_(ツ)_/¯

File: views/templates/nav.ejs [hunk 6]
```diff
@@ -65,15 +65,20 @@
...
-
         </li>
         <li class="list-group-item<% if (locals.page !== undefined && page == 'autoscaling') { %> active<% } %>">
             <a href="/autoscaling">
-                <span class="fa fa-cubes" data-toggle="tooltip" title="Auto Scaling"></span>
-                <span class="list-group-item-value">Auto Scaling</span>
+                <span class="fa fa-cubes" data-toggle="tooltip" title="Pod Auto Scaling"></span>
+                <span class="list-group-item-value">Pod Auto Scaling</span>
+            </a>
+        <% if (locals.cloud === "AWS" && cloud) { %>
```
Comment (wgordon17) [inline views/templates/nav.ejs]: This seems...backward 🤔 Where is `cloud` getting defined? I think `page` is a global EJS kinda thing, but not sure about `cloud` ¯\\\_(ツ)_/¯

**Review by wgordon17 — COMMENTED** [review]
Really great! Looks good Oren, just threw a couple of comments out 👍

File: app.js (hunk 1 above)
Comment (0kashi) [reply app.js]: Since S3 is global it doesn't matter much for this specific use case. I can probably remove the comment.

File: README.md (hunk 2 above)
Comment (0kashi) [reply README.md]: That would be ideal.  I'll look into this and thanks for the link.

File: app.js (hunk 3 above)
Comment (0kashi) [reply app.js:302]: Well, without providing a key and secret, yes.  It is using STS though.

File: views/filesystem.ejs (hunk 5 above)
Comment (0kashi) [reply views/filesystem.ejs]: Good point!  I'll change the link.

File: views/templates/nav.ejs (hunk 6 above)
Comment (0kashi) [reply views/templates/nav.ejs]: corrected.

File: README.md (hunk 2 above)
Comment (0kashi) [reply README.md]: As we reviewed this approach is not easily implemented.

File: README.md [hunk 7]
```diff
@@ -33,6 +34,8 @@ the ConfigMap object
...
+- **NAMESPACE**: Namespace that this application is deployed into. This is used for the S3 bucket.
```
Comment (wgordon17) [inline README.md]: Still needed?

**Review by wgordon17 — APPROVED** [review]
This looks great!

File: README.md (hunk 7 above)
Comment (0kashi) [reply README.md]: Nope.  I removed it in the latest version.

**Review by wgordon17 — APPROVED** [review]
LGTM

**Pull request merged**